lua/ido.lua

```diff
@@ -138,42 +139,19 @@ end
 -- }}}
 -- Get the matching items -{{{
 function ido_get_matches()
-
   local ido_pattern_text, true_ido_pattern_text = ido_pattern_text, ido_pattern_text
   ido_matched_items, ido_current_item = {}, ""
   local ido_true_matched_items = {}
 
-  if ido_fuzzy_matching then
-    ido_pattern_text = ido_pattern_text:gsub('.', '.*%1')
+  if not ido_fuzzy_matching then
```

Comment:
This option should probably be removed as `fzy` does not support NON-fuzzy matching.

Reply:
That won't be a problem. We have to do a non-fuzzy match on each item in the filtered list for figuring out the prefix anyways. So if `ido_fuzzy_matches` is `false`, then just remove the items where the above function returns nil, since prefixes are only calculated on true-matched items. So something like

```lua
for key, item in pairs(items) do
   if string.find(item, pattern, 1, true) then
      -- Calculate prefix
   else
      if not ido_fuzzy_matching then
         table.remove(items, key)
      end
   end
end
```